Enigmatry.Entry.AspNetCore.Authorization/AuthorizationStartupExtensions.cs

```diff
@@ -0,0 +1,36 @@
+﻿using Enigmatry.Entry.AspNetCore.Authorization.Requirements;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Builder;
+using Microsoft.Extensions.DependencyInjection;
+
+namespace Enigmatry.Entry.AspNetCore.Authorization;
+public static class AuthorizationStartupExtensions
+{
+    public static ControllerActionEndpointConventionBuilder AppRequireAuthorization(
+        this ControllerActionEndpointConventionBuilder builder,
+        bool enable) =>
+        !enable ? builder : builder.RequireAuthorization();
+
+    public static AuthorizationBuilder AppAddAuthorization(this IServiceCollection services, bool enable)
+    {
+        if (!enable)
+        {
+            return services.AddAuthorizationBuilder();
+        }
+
+        services.AddScoped<IAuthorizationHandler, UserHasRole.RequirementHandler>();
+        services.AddScoped<IAuthorizationHandler, UserHasPermission.RequirementHandler>();
+
+        return services.AddAuthorizationBuilder()
+            .AddPolicy(
+                PolicyNames.UserHasRole,
+                policy => policy.AddRequirements(new UserHasRole.Requirement()))
```

Comment:
The more I look code, the less I see need for role-based policy. Permissions are core and role can evolve during time so role-based might need to be optional here.

Reply:
I think that is the nice the way it is set up now. The actual implementation of the rol based policy can be done by the application implementing the library, by implementing IAuthorizationProvider. Could be based on claims, or database role tables, or something completely defferent.